File: data/multilang_comparison/program_cpp_vector/src/func21.c
#include "program_cpp_vector.h" 
array_t* func21(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array173;
   if (pCounter > 0) {
      array173 = vars->data[--pCounter];
      array173->refC++;
      DEBUG_COPY(array173->id);
   } else {
      array173 = (array_t*)malloc(sizeof(array_t));
      array173->size = 189;
      array173->refC = 1;
      array173->id = 173;
      array173->data = (unsigned int*)malloc(array173->size*sizeof(unsigned int));
      memset(array173->data, 0, array173->size*sizeof(unsigned int));
      DEBUG_NEW(array173->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array173;
   array_t* array174 = func23(&params0, loopsFactor);
   DEBUG_RETURN(array174->id);
   free(params0.data);
   if(PATH0 & 1) {
      array_t* array175;
      if (pCounter > 0) {
         array175 = vars->data[--pCounter];
         array175->refC++;
         DEBUG_COPY(array175->id);
      } else {
         array175 = (array_t*)malloc(sizeof(array_t));
         array175->size = 641;
         array175->refC = 1;
         array175->id = 175;
         array175->data = (unsigned int*)malloc(array175->size*sizeof(unsigned int));
         memset(array175->data, 0, array175->size*sizeof(unsigned int));
         DEBUG_NEW(array175->id);
      }
      unsigned int loop51 = 0;
      unsigned int loopLimit51 = (rand()%loopsFactor)/3 + 1;
      for(; loop51 < loopLimit51; loop51++) {
         for (int i = 0; i < array175->size; i++) {
            array175->data[i]--;
         }
         for (int i = 0; i < array173->size; i++) {
            if (array173->data[i] == 87) { 
               return array173;
            }
         }
      }
      array_t_param params1;
      params1.size = 3;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array173;
      params1.data[1] = array174;
      params1.data[2] = array175;
      array_t* array176 = func36(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array176->id);
      free(params1.data);
      array176->refC--;
      if(array176->refC == 0) {
         free(array176->data);
         free(array176);
         DEBUG_FREE(array176->id);
      }
      array175->refC--;
      if(array175->refC == 0) {
         free(array175->data);
         free(array175);
         DEBUG_FREE(array175->id);
      }
   }
   else {
      array_t_param params1;
      params1.size = 2;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array173;
      params1.data[1] = array174;
      array_t* array177 = func32(&params1, get_path(), loopsFactor);
      DEBUG_RETURN(array177->id);
      free(params1.data);
      array177->refC--;
      if(array177->refC == 0) {
         free(array177->data);
         free(array177);
         DEBUG_FREE(array177->id);
      }
   }
   array174->refC--;
   if(array174->refC == 0) {
      free(array174->data);
      free(array174);
      DEBUG_FREE(array174->id);
   }
   return array173;
}

File: data/multilang_comparison/program_go/program_go/src/func21.go
package main
import "math/rand"
var _ = rand.Intn
func Func21(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array173 *Array
   if pCounter > 0 {
       array173 = vars.Data[pCounter-1]
       pCounter--
       array173.RefC++
   } else {
       array173 = &Array{
           Size: 189,
           RefC: 1,
           Id: 173,
           Data: make([]uint32, 189),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array173
   var array174 *Array = Func23(&params0, loopsFactor);
   params0.Data = nil
   params0.Size = 0
   if (PATH0 & 1) != 0 {
      var array175 *Array
      if pCounter > 0 {
          array175 = vars.Data[pCounter-1]
          pCounter--
          array175.RefC++
      } else {
          array175 = &Array{
              Size: 641,
              RefC: 1,
              Id: 175,
              Data: make([]uint32, 641),
          }
      }
      var loop51 int = 0;
      var loopLimit51 int = (50)/3 + 1;
      for ; loop51 < loopLimit51; loop51++ {
         for i := 0; i < array175.Size; i++ {
             array175.Data[i]--
         }
         for i := 0; i < array173.Size; i++ {
             if array173.Data[i] == 87 {
                 return array173
             }
         }
      }
      var params1 ArrayParam
      params1.Size = 3
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array173
      params1.Data[1] = array174
      params1.Data[2] = array175
      var array176 *Array = Func36(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array176.RefC--
      if array176.RefC == 0 {
          array176.Data = nil
          array176 = nil
      }
      array175.RefC--
      if array175.RefC == 0 {
          array175.Data = nil
          array175 = nil
      }
   
   } else {
      var params1 ArrayParam
      params1.Size = 2
      params1.Data = make([]*Array, params1.Size)
      params1.Data[0] = array173
      params1.Data[1] = array174
      var array177 *Array = Func32(&params1, getPath(), loopsFactor);
      params1.Data = nil
      params1.Size = 0
      array177.RefC--
      if array177.RefC == 0 {
          array177.Data = nil
          array177 = nil
      }
   }
   array174.RefC--
   if array174.RefC == 0 {
       array174.Data = nil
       array174 = nil
   }
   return array173;
}

